api/api_test.go

```diff
@@ -0,0 +1,446 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/beme/abide"
+	golog "github.com/ipfs/go-log"
+	"github.com/qri-io/dataset/dsfs"
+	"github.com/qri-io/qri/config"
+	"github.com/qri-io/qri/lib"
+	"github.com/qri-io/qri/repo"
+	"github.com/qri-io/qri/repo/profile"
+	"github.com/qri-io/qri/repo/test"
+	regmock "github.com/qri-io/registry/regserver/mock"
+)
+
+func init() {
+	abide.SnapshotsDir = "testdata"
+}
+
+func newTestRepo(t *testing.T) (r repo.Repo, teardown func()) {
+	var err error
+	if err = confirmQriNotRunning(); err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	// bump up log level to keep test output clean
+	golog.SetLogLevel("qriapi", "error")
+
+	// use a test registry server & client & client
+	rc, registryServer := regmock.NewMockServer()
+	// we need to artificially specify the timestamp
+	// we use the dsfs.Timestamp func variable to override
+	// the actual time
+	prevTs := dsfs.Timestamp
+	dsfs.Timestamp = func() time.Time { return time.Date(2001, 01, 01, 01, 01, 01, 01, time.UTC) }
+
+	if r, err = test.NewTestRepo(rc); err != nil {
+		t.Fatalf("error allocating test repo: %s", err.Error())
+	}
+
+	lib.Config = config.DefaultConfig()
```

Comment:
Please use DefaultConfigForTesting from config/test_config.go : https://github.com/qri-io/qri/blob/31be72d4e30742ac7ea731600f66f769ead3afb1/config/test_config.go#L8

This avoids generating private keys, using pregenerated keys instead.